### DISCUSSION TITLE
Bulk secret creation via single API call

### DISCUSSION THREAD

**Discussion by xeronix** [fabric8io/kubernetes-client#5820] [General]
Hello,

We need to create multiple K8s secrets. As of now we are creating each secret via separate API call using fabric8io k8s client.
How can we create multiple secrets via single API call ?

**Comment by rohanKanojia** [comment]
Try using the `resourceList()` DSL entrypoint to provide SecretList which contains the list of Secrets you want to create.

https://github.com/fabric8io/kubernetes-client/blob/464e32eb7c8c05acd3ecd2fa90c0a295d6e80a59/kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/ResourceListTest.java#L112

**Reply by xeronix** [reply]
This does not help. It still makes multiple HTTP calls to API server.
I thing API itself is not available.
We do have bulk delete API -> https://pwittrock.github.io/docs/api-reference/v1.7/#delete-collection-205
But, not able to find bulk create API.

**Reply by manusa** [reply]
I don't think the Kube API supports bulk resource creation:

https://kubernetes.io/docs/reference/using-api/api-concepts/#single-resource-api

> ## Single resource API
>
> The Kubernetes API verbs get, create, update, patch, delete and proxy support single resources only. These verbs with single resource support have no support for submitting multiple resources together in an ordered or unordered list or transaction.

So, from the Fabric8 Kubernetes Client standpoint, it's impossible to achieve what you want.

**Reply by xeronix** [reply]
Yes, will raise a request for this API in K8s project itself.
